src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -1173,7 +1119,14 @@ public function isNotFound()
      */
     public function isRedirect($location = null)
     {
-        return in_array($this->statusCode, array(201, 301, 302, 303, 307, 308)) && (null === $location ?: $location == $this->headers->get('Location'));
+        return in_array($this->statusCode, array(
+            self::HTTP_CREATED,
+            self::HTTP_MOVED_PERMANENTLY,
+            self::HTTP_FOUND,
+            self::HTTP_SEE_OTHER,
+            self::HTTP_TEMPORARY_REDIRECT,
+            self::HTTP_PERMANENTLY_REDIRECT
```

Comment:
So, developers have asked for constants, but because these constants are not a source of truth within the class, it is conceivable that, due to Symfony contributor error, I could use an `HTTP_` constant, and that integer does not coincide with the internal manifestation? That sounds like a terribly hard thing to debug if it happened.

I'm still struggling with the concept of having two variables (and I use the term to include constants) that are supposed to mean the same thing, but are individually set up. Having them as one and the same would make there one point of error if there was an error. It sounds from your description as if the constants are unverified; I just find it weird that you would expose an unverified integer. I would always assume that `HTTP_OK` === the corresponding key in the `statusClass` array; instead, it is actually == because they are _not_ the same.